src/bitreader.cpp

```diff
@@ -16,16 +18,23 @@ BitReader::BitReader(FILE* file)
 
 uint8_t BitReader::read_unaligned_byte()
 {
-    auto c = getc(file);
-    if (c == EOF)
-        // TODO: Throw a specified for this case value (not just 0)
-        throw 0;
+    //auto c = getc(file);
+    char c;
+    lzmaread(&c, sizeof(char), 1, file);
```

Comment:
You could check against the return type, which should represent the read count. If it does not equal 1, then an error probably occured or EOF happened. Thats at least the behaviour of the standard function fread: 
http://www.cplusplus.com/reference/cstdio/fread/